src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+
+import { Todo } from '../types/Todo';
+
+type Props = {
+  isTodoShow: boolean,
+  newTodoTitle: string,
+  handleNewTodoTitle: (
+    newTodoTitle: React.ChangeEvent<HTMLInputElement>
+  ) => void,
+  addNewTodo: (title: string) => Promise<Todo | null>,
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  isTodoShow,
+  newTodoTitle,
+  handleNewTodoTitle,
+  addNewTodo,
+}) => {
+  const [todoAdd, setTodoAdd] = useState(false);
+
+  const submitForm = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setTodoAdd(true);
+
+    await addNewTodo(newTodoTitle);
```

Comment:
Missing error handling for the promise returned by addNewTodo function. It is best practice to handle errors for all promises to prevent unexpected crashes. You should use try/catch for error handling here.